src/ejabberd_auth.erl

```diff
@@ -101,6 +111,78 @@ plain_password_required(Server) ->
               M:plain_password_required()
       end, auth_modules(Server)).
 
+-spec set_generic_opts(Host :: ejabberd:server(),
+                      Opt :: atom(),
+                      KVs :: [tuple()]) ->  {atomic|aborted, _}.
+set_generic_opts(Host, Opt, KVs) ->
+    OldOpts = ejabberd_config:get_local_option(Opt, Host),
+    AccFunc = fun({K, V}, Acc) ->
+                  lists:keystore(K, 1, Acc, {K, V})
+              end,
+    NewOpts = lists:foldl(AccFunc, OldOpts, KVs),
+    ejabberd_config:add_local_option({Opt, Host}, NewOpts).
+
+-spec get_generic_opt(Host :: ejabberd:server(),
+                      Opt :: atom(),
+                      Key :: ejabberd:key()) -> undefined | ejabberd:value().
+get_generic_opt(Host, Opt) ->
```

Comment:
Is it that useful to fetch all options? I would rather expect an API like `get_opt(Host, Opt)` and `get_opt(Host, Opt, Default)`.